flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -861,14 +861,18 @@ public void grantLeadership(final UUID newLeaderSessionID) {
 			getMainThreadExecutor());
 	}
 
-	protected CompletableFuture<Void> getJobTerminationFuture(JobID jobId) {
+	CompletableFuture<Void> getJobTerminationFuture(JobID jobId) {
 		if (jobManagerRunners.containsKey(jobId)) {
 			return FutureUtils.completedExceptionally(new DispatcherException(String.format("Job with job id %s is still running.", jobId)));
 		} else {
 			return jobManagerTerminationFutures.getOrDefault(jobId, CompletableFuture.completedFuture(null));
 		}
 	}
 
+	CompletableFuture<Void> getRecoveryOperation() {
```

Comment:
This method has wider visibility scope than necessary, and is part of production code. I think `@VisibleForTesting` should be added.

Reply:
Good point. Will add it.